### PR TITLE
Static graph

### PR DESCRIPTION
**Pull request by PeterSH6**
### Overview
- Implement GraphSage and GAT using dgl.nn
- GAT doesn't have official edge prediction implementation, so I use the same edge predictor as graphsage. And the network architecture and hyperparameter only have very little differences between node classification
- Creating a new training script and maybe we need to merge them together to a single training script.
### Sampling
- For Sampling, just use uniform sampling and timestamp array is filled with `np.finfo(np.float32).max))`
- Testing the accuracy using REDDIT and synthetic node features.
- The Hyper-Parameter may need some modification. GraphSage has 3 layers (fanouts: [15, 10, 5]) and GAT has 2 layers (fanouts: [10, 5], heads=[8, 1]), lr = 0.0005, batchsize=600, dim_out = 100.

| model | epoch 1  | epoch 2 | epoch 3 | 
|  ----  | ----  | ----  | ----  |
| graphsage  | 98.1 | 98.6 | 98.9 | 
| gat | 94.6 | 96.2 | 96.7 |

### CODE REVIEW

**Comment by PeterSH6** [comment]
Delete the `offline_static_edge_prediction.py` file.
Now, sampler will have a flag to indicate if it is a static graph sampling request

**Review by jasperzhong — APPROVED** [review]
